jpa/subsystem/src/main/java/org/jboss/as/jpa/processor/JPADelegatingClassFileTransformer.java

```diff
@@ -73,7 +73,8 @@ public ByteBuffer run() {
                                 // because we won't ever be called to transform already loaded classes.
                                 result = transformer.transform(classLoader, className, null, protectionDomain, transformedBuffer);
                             } catch (Exception e) {
-                                throw JpaLogger.ROOT_LOGGER.invalidClassFormat(e, className);
+                                ROOT_LOGGER.error(JpaLogger.ROOT_LOGGER.invalidClassFormat(className),e);
+                                throw JpaLogger.ROOT_LOGGER.invalidClassFormat(className);
```

Comment:
A log and rethrow with the same string content is generally an anti-pattern, so please precede this with a comment explaining why it's done. AIUI the reason is for some reason the exception handling higher in the stack ends up not logging the original 'e' cause's stack, and the code that doesn't log it is not in this source repo, so you log it here. But I had to scroll through the long comments on the JIRA to figure that out, so it should be explained here. :)

Reply:
A comment could be:

```
// ModuleClassLoader.defineClass discards the cause of the exception we throw, so to ensure it is logged we log it here.
```

BTW the relevant code is ModuleClassLoader L488-496 which creates a new ClassFormatError from the one thrown at L470, but discards the L470 one's 'cause' field.

@ropalka @dmlloyd is the discarding of the cause deliberate or just an oversight? It looks like the purpose of that code is just to tweak the message.

Reply:
> @ropalka @dmlloyd is the discarding of the cause deliberate or just an oversight? It looks like the purpose of that code is just to tweak the message.

I think it's an oversight. We should retain that information.

Reply:
Thanks @dmlloyd -- I filed https://issues.redhat.com/browse/MODULES-442